### ISSUE TITLE
shortcut to other providers (bitbucket/gitlab/...) doesn't work

### ISSUE THREAD

**Issue by huochunpeng** [pnpm/git-resolver#10]
Currently shortcut only works for github hosted repo, like `username/repo` or `github:username/repo`.

Current implementation assumes we only use shortcut for github repos. When I try anything like

```
pnpm install bitbucket:buttonwoodcx/bcx-ui#v1.14.20
```

It got error in pnpm-debug.log:
```
ERROR  Command failed: git ls-remote --refs 
fatal: 'null' does not appear to be a git repository
fatal: Could not read from remote repository.
```

Note, current pnpm manifests the error in terminal as
```
err.code.startsWith is not a function
```

Which is another issue. When this fails, somehow `err.code` is a number, not string. Need some fix in `pnpm/packages/default-reporter/src/reportError.ts`.

-----
I am trying to add support of shortcut to other providers, kind of works in local test code. But I have trouble to setup my dev environment to force my local global installed `pnpm` to use my local `git-resolver`.
`pnpm link` does not work for `git-resolver`, I guess it's because the global installed `pnpm` package maintained its own node_modules, not refering to global installed `git-resolver`.

I would appreciate if someone gives me some guide on how to setup dev environment for this. Thx.

**Comment by zkochan** [comment]
@huochunpeng the easiest solution would be just to add a new test to this repo for bitbucket. There is already good coverage: https://github.com/pnpm/git-resolver/blob/master/test/index.ts

When you fix it in this repo, I will publish a new version of it and you'll be able to test it in the pnpm monorepo: https://github.com/pnpm/pnpm

**Comment by zkochan** [comment]
if you need assistance, you can also write in our dev chatroom: https://gitter.im/pnpm/development

**Comment by huochunpeng** [comment]
closed by #12
